Add custom exceptions md #5788 (#5833)
Codecademy/docs: content/python/concepts/errors/errors.md

ratio = 100 / 0
ZeroDivisionError: division by zero
```

## Custom Exceptions

> Custom exceptions can be used to define and handle application-specific errors. This can be done by creating a sub-class of the built-in `Exception` class. Custom exceptions are useful for providing descriptive user feedback for unexpected behaviour. They can also help developers in debugging or be used elsewhere in the program. 

Reviewer: The last 2 sentences can be merged together for better readability.

```suggestion
Custom exceptions can be used to define and handle application-specific errors. This can be done by creating a sub-class of the built-in `Exception` class. 

Custom exceptions help provide clear feedback to users about unexpected behavior, aid in debugging, and can be used throughout the program. 
```

### Defining Custom Exceptions

To define a custom exception, create a new class that inherits from `Exception` and optionally override the constructor to provide additional information.

```pseudo
#Custom Exception for invalid age input
class InvalidAgeError(Exception):
    def __init__(self, message, age=None):
        super().__init__(message)
        self.age = age
```

### Using Custom Exceptions

> Once a custom exception has been created, it can be used to raise custom error messages for invalid inputs. For example, this function raises an error if the user inputs an age not between 0 and 130.

Reviewer: ```suggestion
Once a custom exception has been created, it can raise custom error messages for invalid inputs. For example, the following function raises an error if the user inputs an age not between 0 and 130.
```

```pseudo
def validate_age(age):
    if not isinstance(age, int):
        raise InvalidAgeError("Age must be an integer.", age)
    if age < 0 or age > 130:
        raise InvalidAgeError(f"{age} is not a valid age. Please choose an age between 0 and 130.", age)
```

### Example Custom Exception Usage

> Once the custom exception has been used in a function, this function can be used to throw the custom error to the user.

Reviewer: It can be rewritten with a little more context as follows:

```suggestion
Once the custom exception is defined, it can be used to handle errors in a `try-except` block. The following example demonstrates how to raise the custom exception when an invalid age is provided.
```
